frontend/src/libraries/explorer-wamp/blocks.ts

```diff
@@ -87,15 +94,21 @@ export default class BlocksApi extends ExplorerApi {
           blockId,
         },
       ]).then((it) => (it[0].hash !== null ? it[0] : null));
-
+      const finalHeight = await this.getFinalStats();
+      block.isFinal = block.height <= finalHeight;
```

Comment:
if `block` is `null`, wouldn't it crash here?

Reply:
good catch

Reply:
It will.

@icerove 

1. Do the final stats fetching in parallel with the select (use `await Promise.all`)
2. Move the assignment after the null check